### DISCUSSION TITLE
Adding fzf-lua providers

### DISCUSSION THREAD

**Discussion by folke** [ibhagwan/fzf-lua#1246] [Q&A] [answered]
Hi!

I just added support for `fzf-lua` to `todo-comments.nvim`.
See [here](https://github.com/folke/todo-comments.nvim/blob/main/lua/todo-comments/fzf.lua)

Right now, I just call that todo method as in `require("todo-comments.fzf").todo()`

However, it would be great if it could also work using `:FzfLua todo`

Is that currently possible to somehow **register** new providers?

**Comment by gennaro-tedesco** [comment]
I have a similar question indeed. In my personal configuration I have a few custom fzf-lua providers that I invoke by defining specific ad-hoc commands, it would however be more consistent to have a way to invoke them simply as `:FzfLua <custom_provider>` (also so that they can show in the completion menus).

**Comment by ibhagwan** [comment, marked as answer]
Interesting, I missed this discussion.

Don’t have a dedicated `register` but I believe that if you add your function to the main package metatable it should show up in `builtin`:
```lua
require('fzf-lua').foo = function(…)
```

I’m not opposed to having a `register` API for extensions (perhaps with name, default settings, etc).

**Reply by gennaro-tedesco** [reply]
Fantastic, it works as intended!

![Kapture 2025-01-13 at 11 35 13](https://github.com/user-attachments/assets/f4c77e37-efc7-4cfb-bb84-66118cb284ab)

If this is the case, I do not think we need a dedicated `register` API (the register API itself being the main package metatable, after all).

**Reply by ibhagwan** [reply]
Ty for the update @gennaro-tedesco!
